src/ads/store.bal

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/random;
+
+readonly class AdStore {
```

Comment:
IMO, rather than having a class we could just use an isolated `map<Ad[]>` as the in-memory store and just implement a public methods to retrieve random ads and retrieve ads per category.

Reply:
Any reason why we shouldn't use the class? I've added this readonly class as a example as a feature in readme as well, if this is bad practice, we have to remove it from there as well.

Reply:
IMOO `functions` are first class citizens in Ballerina. Hence we did not have to instantiate objects to invoke functions. It reduces the number of lines we have to write to implement a function. It does not mean using classes is a bad practice though.